InterfacePython/TasmanianParticleSwarm.py

```diff
@@ -0,0 +1,264 @@
+        if (llfNewPPosns.shape[1] != iNumDims):
+            raise InputError("llfNewPPosns", "llfNewPPosns.shape[1] should match the number of dimensions")
+        pLibDTSG.tsgParticleSwarmState_SetParticlePositions(self.pStatePntr, llfNewPPosns)
+
+    def setParticleVelocities(self, llfNewPVelcs):
+        '''
+        Set new particle velocities from a NumPy array.
+
+        llfNewPVelcs : a two-dimensional numpy.ndarray with
+            .shape[0] = .getNumParticles()
+            .shape[1] = .getNumDimensions()
+        '''
+        iNumPart = self.getNumParticles()
+        iNumDims = self.getNumDimensions()
+        if (llfNewPVelcs.shape[0] != iNumPart):
+            raise InputError("llfNewPVelcs", "llfNewPVelcs.shape[0] should match the number of particles")
+        if (llfNewPVelcs.shape[1] != iNumDims):
+            raise InputError("llfNewPVelcs", "llfNewPVelcs.shape[1] should match the number of dimensions")
+        pLibDTSG.tsgParticleSwarmState_SetParticleVelocities(self.pStatePntr, llfNewPVelcs)
+
+    def setBestParticlePositions(self, llfNewBPPosns):
+        '''
+        Set new best particle positions from a NumPy array.
+
+        llfNewBPPosns : a two-dimensional numpy.ndarray with
+            .shape[0] = .getNumParticles() + 1
+            .shape[1] = .getNumDimensions()
+        '''
+        iNumPart = self.getNumParticles()
+        iNumDims = self.getNumDimensions()
+        if (llfNewBPPosns.shape[0] != iNumPart + 1):
+            raise InputError("llfNewBPPosns", "llfNewBPPosns.shape[0] should match the number of particles + 1")
+        if (llfNewBPPosns.shape[1] != iNumDims):
+            raise InputError("llfNewBPPosns", "llfNewBPPosns.shape[1] should match the number of dimensions")
+        llfNewBPPosns.resize(((iNumPart + 1) * iNumDims,))
+        pLibDTSG.tsgParticleSwarmState_SetBestParticlePositions(self.pStatePntr, llfNewBPPosns)
+
+    def clearBestParticles(self):
+        '''
+        Clears the vector of best particles.
+        '''
+        pLibDTSG.tsgParticleSwarmState_ClearBestParticles(self.pStatePntr)
+
+    def clearCache(self):
+        '''
+        Clears the internal cache of the managed ParticleSwarm C++ object.
+        '''
+        pLibDTSG.tsgParticleSwarmState_ClearCache(self.pStatePntr)
+
+    def initializeParticlesInsideBox(self, lfBoxLower, lfBoxUpper, random01 = DREAM.RandomGenerator(sType = "default")):
+        '''
+        Initialize the particle swarm state with randomized particles (determined by gen_random01) inside a box bounded by the
+        parameters box_lower and box_upper. The i-th entry in these parameters respectively represent the lower and upper
+        bounds on the particle position values for the i-th dimension. The parameter random01 controls the distribution of
+        the particles.
+
+        lfBoxLower : a one-dimensional NumPy array with lfBoxLower.shape[0] = .getNumDimensions()
```

Comment:
Why the `.` notation, the method is either `self.getNumDimensions()` or simply `getNumDimensions()`.

Reply:
This is to stay consistent with the other DREAM modules (e.g., see the documentation for `TasmanianDreamState.setState()`). I'll fix it to one of the above in the next push.

Reply:
Done.

Reply:
Change it to `self` here, the other one should also have `self` but the `.shape` can stay.

Reply:
Done.